core/errors/infer.h

```diff
@@ -31,5 +31,6 @@ constexpr ErrorClass ProcArityUnknown{7023, StrictLevel::Strict};
 constexpr ErrorClass GenericPassedAsBlock{7024, StrictLevel::True};
 constexpr ErrorClass AbstractClassInstantiated{7025, StrictLevel::True};
 constexpr ErrorClass NotExhaustive{7026, StrictLevel::True};
+constexpr ErrorClass GenericTypeParamBoundMismatch{7027, StrictLevel::True};
```

Comment:
I think you want this in `false` as the intrinsic is used to interpred `sig`s in untyped files too

Reply:
Also, can we add a test for that? A generic with incorrect bounds used in `sig` in an untyped file

Reply:
I'm not sure we can, as T_Generic_squareBrackets doesn't appear to be executed when the file is marked `typed: false`.

Reply:
Ah, I misread the part in your original comment about `sig`s. I've made this change and added a test.